src/lib/Tests/Service/PermissionAwareLocationResolverTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Tests\Service;
+
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\Repository\Values\Content\Location;
+use EzSystems\EzPlatformAdminUi\Service\PermissionAwareLocationResolver;
+use PHPUnit\Framework\TestCase;
+use eZ\Publish\API\Repository\LocationService;
+
+class PermissionAwareLocationResolverTest extends TestCase
+{
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Service\LocationResolver */
+    private $locationResolver;
+
+    public function setUp()
+    {
+        $this->locationService = $this->createMock(LocationService::class);
+
+        $this->locationResolver = new PermissionAwareLocationResolver($this->locationService);
+    }
+
+    /**
+     * Test for the resolveLocation() method.
+     *
+     * @see \EzSystems\EzPlatformAdminUi\Service\PermissionAwareLocationResolver::resolveLocation()
+     */
+    public function testResolveMainLocation()
+    {
+        $contentInfo = new ContentInfo(['mainLocationId' => 42]);
+        $location = new Location(['id' => 42]);
+
+        // User has access to the main Location
+        $this->locationService
+            ->expects($this->once())
```

Comment:
I don't think the essence of this test is to check the number of calls. Unless there's a specific reason, you can skip it.